clients/src/test/java/org/apache/kafka/common/record/AbstractLegacyRecordBatchTest.java

```diff
@@ -56,6 +57,44 @@ public void testSetLastOffsetCompressed() {
             assertEquals(offset++, record.offset());
     }
 
+    @Test
+    public void testIterateCompressedRecordWithWrapperOffsetZero() {
```

Comment:
Maybe worth adding a comment saying that this is unexpected for V1, but librdkafka does it.